spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java

```diff
@@ -870,6 +899,10 @@ private boolean rawByParameterIsType(Type parameterType, Type type) {
 		return parameterType instanceof ParameterizedType pType && pType.getRawType().equals(type);
 	}
 
+	public void setObservationRegistry(ObservationRegistry observationRegistry) {
```

Comment:
The setter is in wrong place.
Has to be somewhere after ctor with the rest of setters.
And the method might deserve JavaDoc since it is `public`.